core/src/replay_stage.rs

```diff
@@ -673,6 +686,57 @@ impl ReplayStage {
         Ok(())
     }
 
+    fn snapshot_tower(tower_snapshot_path: &PathBuf, tower: &Tower) -> Result<()> {
+        let timer = Instant::now();
+        fs::create_dir_all(tower_snapshot_path)?;
+        let mut snapshot_file = File::create(tower_snapshot_path.join(TOWER_SNAPSHOT_NAME))?;
+        snapshot_file.write_all(&bincode::serialize(&tower).expect("tower serialize failed"))?;
+        snapshot_file.flush()?;
+        let snapshot_time = timer.elapsed().as_millis() as usize;
+        inc_new_counter_info!("replay_stage-tower_snapshot_duration_ms", snapshot_time);
+        Ok(())
+    }
+
+    fn reload_tower(
+        my_pubkey: &Pubkey,
+        tower_snapshot_path: &PathBuf,
+        vote_account: &Pubkey,
+        bank_forks: &Arc<RwLock<BankForks>>,
+    ) -> Tower {
+        info!("Looking for Tower snapshot at {:?}", tower_snapshot_path);
```

Comment:
Seems like a better place would be to put this save/restore is where Tower lives in `consensus.rs`

Reply:
Yup!